### ISSUE TITLE
Issue on labels plotting - Same labels not appear

### ISSUE THREAD

**Issue by xellDart** [ultralytics/yolov5#10773]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Multiple annotations do not appear in the generated set, Im using roboflow and export to yolov5 pytorch, but I dont understand why some annotation not appear in batches.

When I run detection on trained model, I have some issues in this part.

Training batch

![imagen](https://user-images.githubusercontent.com/6379241/212521993-e914efa8-c2a2-488f-8aed-05d34ab9ce11.png)
![imagen](https://user-images.githubusercontent.com/6379241/212522002-83a36660-d855-49f8-b65d-7943109fac98.png)
![imagen](https://user-images.githubusercontent.com/6379241/212522012-670ef6c6-31a2-4899-91c3-b957ca73fe47.png)
![imagen](https://user-images.githubusercontent.com/6379241/212523890-ede189a3-bf37-4615-91f3-55131123aeda.png)
![imagen](https://user-images.githubusercontent.com/6379241/212522017-19b8473b-fd13-443c-9ce0-bca8e9b56955.png)
![imagen](https://user-images.githubusercontent.com/6379241/212523927-a11ce462-9872-40ff-b477-c0fce0e39153.png)

Roboflow
Im annotated las 3 lines

<img width="1103" alt="Captura de pantalla 2023-01-14 a la(s) 23 19 17" src="https://user-images.githubusercontent.com/6379241/212524167-e1d3616d-0134-4e7e-967b-a8c490e9ca18.png">

Training inference using yolov5l

![imagen](https://user-images.githubusercontent.com/6379241/212522063-ce953dee-739a-41af-a8b9-a5c42850b35a.png)

Im training same dataset using coco on yolox and nanodet, but i dont have this issue

### Environment

Yolo v5 from master branch

Pytorch 1.13.1+cu117

### Minimal Reproducible Example

Training with standard configuration

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by xellDart** [comment]
I found that some  boxes are skipped on

```
def box_candidates(box1, box2, wh_thr=2, ar_thr=20, area_thr=0.2, eps=1e-16):  # box1(4,n), box2(4,n)
    # Compute candidate boxes: box1 before augment, box2 after augment, wh_thr (pixels), aspect_ratio_thr, area_ratio
    w1, h1 = box1[2] - box1[0], box1[3] - box1[1]
    w2, h2 = box2[2] - box2[0], box2[3] - box2[1]
    ar = np.maximum(w2 / (h2 + eps), h2 / (w2 + eps))  # aspect ratio
    return (w2 > wh_thr) & (h2 > wh_thr) & (w2 * h2 / (w1 * h1 + eps) > area_thr) & (ar < ar_thr)  # candidates
```

I am adjusting manually allowing the allowed value to open more, my question is, can this affect the performance of the model?

**Comment by glenn-jocher** [comment]
@xellDart the `box_candidates` function you're adjusting filters candidate boxes based on their width, height, aspect ratio, and area ratio.  By manually adjusting thresholds, you can impact which boxes are considered for training, potentially influencing model performance. Each application is unique and may require fine-tuning, but consider seeking a balanced trade-off between recall and precision. If you have more questions or need further assistance, feel free to ask.
